walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -128,46 +150,137 @@ private boolean attendeesIntersect(
     return false;
   }
 
+  /**
+   * Inserts a time range into a SortedMap containing an ordered overlapping set of times.
+   *
+   * @param rawSet The set of times
+   * @param timeRange The range to insert
+   */
+  private void insertTimeRange(SortedMap<Integer, TimeRange> rawSet, TimeRange timeRange) {
+    if (rawSet.containsKey(timeRange.start())) {
+      // If an event with the same start time is in the set,
+      // keep the longest event
+      TimeRange setTime = rawSet.get(timeRange.start());
+      if (timeRange.duration() > setTime.duration()) {
+        rawSet.put(timeRange.start(), timeRange);
+      }
+    } else {
+      rawSet.put(timeRange.start(), timeRange);
+    }
+  }
+
   /**
    * Converts a conflict set which may contain overlapping intervals into one that contains none.
    * Maximum runtime O(E).
    *
    * @param rawConflictSet
    * @return an ordered conflict set with no overlapping invervals.
    */
-  private Collection<TimeRange> cleanConflictSet(Collection<TimeRange> rawConflictSet) {
+  private List<TimeRange> cleanConflictSet(Collection<TimeRange> rawConflictSet) {
```

Comment:
Can you share why you changed this from a `Collection` to a `List`? No changes needed, just curious. :)

Reply:
So I can implement `combineConflictSets` by using a "standard" merge. The equivalent code with `Collection<TimeRange>` would be a bit hairy for my taste since Java iterators don't have a `peek()` method.